Update Caddy examples and outdated Jitsi docker env variable/documentation link (#2162)
spantaleev/matrix-docker-ansible-deploy: docs/configuring-playbook-own-webserver.md

1) Make sure your web server user (something like `http`, `apache`, `www-data`, `nginx`) is part of the `matrix` group. You should run something like this: `usermod -a -G matrix nginx`. This allows your webserver user to access files owned by the `matrix` group. When using an external nginx webserver, this allows it to read configuration files from `/matrix/nginx-proxy/conf.d`. When using another server, it would make other files, such as `/matrix/static-files/.well-known`, accessible to it.

2) Edit your configuration file (`inventory/host_vars/matrix.<your-domain>/vars.yml`)
   - to disable the integrated nginx server:

        ```yaml
        matrix_nginx_proxy_enabled: false
        ```
    - if using an external server on another host, add the `<service>_http_host_bind_port` or `<service>_http_bind_port` variables for the services that will be exposed by the external server on the other host. The actual name of the variable is listed in the `roles/<service>/defaults/vars.yml` file for each service. Most variables follow the `<service>_http_host_bind_port` format.
       

>       These variables will make Docker expose the ports to the local network instead of localhost only.

Reviewer: ```suggestion
      These variables will make Docker expose the ports on all network interfaces instead of localhost only.
```

      [Keep in mind that there are some security concerns if you simply proxy everything.](https://github.com/matrix-org/synapse/blob/master/docs/reverse_proxy.md#synapse-administration-endpoints)

      Here are the variables required for the default configuration (Synapse and Element)
       ```
        matrix_synapse_container_client_api_host_bind_port: '8008'
        matrix_synapse_container_federation_api_plain_host_bind_port: '8048'

>         matrix_client_element_container_http_host_bind_port: "8765"

Reviewer: Why do you need to tell people to do this explicitly?

We already do it automatically when `matrix_nginx_proxy_enabled: false`.

https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/06161073468027361585b9c850f8975ff71139d2/group_vars/matrix_servers#L2293-L2298

https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/06161073468027361585b9c850f8975ff71139d2/group_vars/matrix_servers#L2300-L2301

https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/06161073468027361585b9c850f8975ff71139d2/group_vars/matrix_servers#L2197-L2200

---

The above also bind to `127.0.0.01:PORT`, instead of just `PORT` (which exposes the port on all interfaces).

---

When someone is using a reverse-proxy on another network, it's probably easier to reconcile that difference by making them share the network (see the `matrix_docker_network` variable).

For reverse-proxying using yet another server (fronting matrix-nginx-proxy), we have these docs too: https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/06161073468027361585b9c850f8975ff71139d2/docs/configuring-playbook-own-webserver.md#method-2-fronting-the-integrated-nginx-reverse-proxy-webserver-with-another-reverse-proxy

Author: You are right, it is not needed when the other webproxy is on a different docker network (same host). I will update my changes to remove that part.

`127.0.0.01:PORT` binds to the loopback interface only. This works only when the other webserver is on the same host (both on `192.158.0.1` for example). There is a good explanation of the difference between 127.0.0.1 and 0.0.0.0 (default for Docker if we don't specifiy an IP address) in this stack overflow : https://stackoverflow.com/a/20778887

If the other webserver is on `192.168.0.2`, it won't be able to reach the service exposed with `127.0.0.1:PORT` on `192.168.0.1`.

Binding to `127.0.0.1` can be a good default, but documenting how to bind the ports to allow another device on the same network to reach the service is necessary. The default in the snippets you linked to could also be changed to bind to `0.0.0.0:PORT` or just `PORT`.

Reviewer: The default try to be safe.

Binding to `0.0.0.0` by default for various services is not great.

Very few people use a reverse-proxy on another machine and need to bind on all interfaces. Most others only need to disable matrix-nginx-proxy and will use another local proxy - fo them, binding to `127.0.0.1` is ideal, and safe.
